src/parser.py

```diff
@@ -333,6 +366,30 @@ def _consume_or_raise(self, token_types, exception_description):
             return self._next_token()
         raise ParseError(token=self._peek(), message=exception_description)
 
+    def _function_call(self, function):
+        """
+        Gathers function and all arguments into a Call expression
+        """
+        arguments = []
+        if not self._is_same_type(TokenType.RIGHT_PAREN):
```

Comment:
If condition inside the loop should probably have an associated break statement. Otherwise it'll process all arguments in excess of 255 but then line 385 will still execute and will try processing closing paren as the last argument, which will throw the parser off.

Reply:
The problem with this change is that the parser will exit the loop before processing all of the arguments and then will try to compare to the right paren, which will not match, and then it will raise the `Expected closing paren ")" at end of function call` error, which is not necessarily correct here.

In the current implementation lines 378-380, we're processing all of the arguments but we're not adding the ones above 255 to the arguments array. That means that we will not save the args over 255, but we'll still process them and discard the result in order for the parser to move past them and be ready to consume the right paren.

Reply:
You were correct about the missing break statement so I've added that.